### PR TITLE
feat(disputer,liquidator,monitorfinancial-templates-lib): rename all instances of emp to financialContract

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

After updating the bots to work with the perpetual, there was a lingering task to rename all instances of `emp` (and associated similar variables) to something more generic that would encompass both the EMP and the perp. This pr does just that. 

The scope of changes is pretty big as a result but I _think_ I hit all key places.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [X]  Existing tests adequate, no new tests required
- [X]  All existing tests pass
- [ ]  Untested

**Note** this is a breaking change within some APIs and will potentially cause issues in the next release. curious what @UMAprotocol/eng thinks about this.

@daywiss are there any other places where I need to change this for affiliates?

### CODE REVIEW

File: packages/affiliates/payouts/dappmining/2021-02-01_2021-02-08_YD-BTC-MAR21_0004.json [hunk 1]
```diff
@@ -5,10 +5,7 @@
     "empAddress": "0x1c3f1A342c8D9591D9759220d114C685FD1cF6b8",
```
Comment (chrismaree) [inline packages/affiliates/payouts/dappmining/2021-02-01_2021-02-08_YD-BTC-MAR21_0004.json:5]: this file was incorrectly linted.

File: packages/affiliates/payouts/devmining/2021-02-01_21-02-02.json [hunk 2]
```diff
@@ -3,80 +3,26 @@
     "startTime": 1612220400000,
```
Comment (chrismaree) [inline packages/affiliates/payouts/devmining/2021-02-01_21-02-02.json:3]: lint

File: packages/disputer/index.js [hunk 3]
```diff
@@ -220,16 +223,16 @@ async function run({
...
-        "Bad input arg! Specify an `EMP_ADDRESS` for the location of the expiring Multi Party within your environment variables."
+        "Bad environment variables! Specify an EMP_ADDRESS of FINANCIAL_CONTRACT_ADDRESS for the location of the financial contract the bot is expected to interact with."
       );
     }
 
     // This object is spread when calling the `run` function below. It relies on the object enumeration order and must
     // match the order of parameters defined in the`run` function.
     const executionParameters = {
-      empAddress: process.env.EMP_ADDRESS,
+      financialContractAddress: process.env.EMP_ADDRESS || process.env.FINANCIAL_CONTRACT_ADDRESS,
```
Comment (chrismaree) [inline packages/disputer/index.js:235]: in the future we should move over to using `FINANCIAL_CONTRACT_ADDRESS` but it should be backward compatible for now.

File: packages/trader/tsconfig.json [hunk 4]
```diff
@@ -4,7 +4,5 @@
   "compilerOptions": {
```
Comment (chrismaree) [inline packages/trader/tsconfig.json:4]: lint

**Review by nicholaspai — APPROVED** [review]
Wow this is a great change, this must have been a PITA to change. If tests pass lgtm

File: packages/financial-templates-lib/src/clients/FinancialContractClient.js [hunk 5]
```diff
@@ -260,7 +260,7 @@ class ExpiringMultiPartyClient {
     });
     this.lastUpdateTimestamp = currentTime;
     this.logger.debug({
-      at: "ExpiringMultiPartyClient",
+      at: "FinancialContractClient",
       message: "Expiring multi party state updated",
```
Comment (daywiss) [inline packages/financial-templates-lib/src/clients/FinancialContractClient.js:264]: does this message need to change?

File: packages/financial-templates-lib/src/clients/FinancialContractEventClient.js [hunk 6]
```diff
@@ -1,35 +1,35 @@
...
+// FinancialContractEventClient to keep a clear separation of concerns and to limit the overhead from querying the chain.
 
-class ExpiringMultiPartyEventClient {
+class FinancialContractEventClient {
   /**
-   * @notice Constructs new ExpiringMultiPartyClient.
+   * @notice Constructs new FinancialContractEventClient.
    * @param {Object} logger Winston module used to send logs.
-   * @param {Object} empAbi Expiring Multi Party truffle ABI object to create a contract instance.
+   * @param {Object} financialContractAbi Expiring Multi Party truffle ABI object to create a contract instance.
```
Comment (daywiss) [inline packages/financial-templates-lib/src/clients/FinancialContractEventClient.js:8]: may need to change comment

File: packages/financial-templates-lib/src/clients/FinancialContractEventClient.js [hunk 7]
```diff
@@ -1,35 +1,35 @@
...
     logger,
-    empAbi,
+    financialContractAbi,
     web3,
-    empAddress,
+    financialContractAddress,
     startingBlockNumber = 0,
     endingBlockNumber = null,
-    contractType = "ExpiringMultiParty", // Default to EMP for now to enable backwards compatibility with other bots. This will be removed as soon as the other bots have been updated to work with these contract types.
+    contractType = "ExpiringMultiParty", // Default to Financial Contract for now to enable backwards compatibility with other bots. This will be removed as soon as the other bots have been updated to work with these contract types.
```
Comment (daywiss) [inline packages/financial-templates-lib/src/clients/FinancialContractEventClient.js:22]: comment might need to say defaults to expiring multi party

File: packages/financial-templates-lib/src/clients/FinancialContractEventClient.js [hunk 8]
```diff
@@ -339,13 +339,13 @@ class ExpiringMultiPartyEventClient {
     this.firstBlockToSearch = lastBlockToSearch + 1;
 
     this.logger.debug({
-      at: "ExpiringMultiPartyEventClient",
+      at: "FinancialContractEventClient",
       message: "Expiring multi party event state updated",
```
Comment (daywiss) [inline packages/financial-templates-lib/src/clients/FinancialContractEventClient.js:343]: does this message need to change?

File: packages/disputer/test/Disputer.js [hunk 9]
```diff
@@ -243,12 +247,12 @@ contract("Disputer.js", function(accounts) {
             transports: [new SpyTransport({ level: "info" }, { spy: spy })]
           });
 
-          // Create a new instance of the ExpiringMultiPartyClient & GasEstimator to construct the disputer
-          empClient = new ExpiringMultiPartyClient(
+          // Create a new instance of the FinancialContractClient & GasEstimator to construct the disputer
+          financialContractClient = new FinancialContractClient(
```
Comment (daywiss) [inline packages/disputer/test/Disputer.js:251]: maybe we can start abbreviating `financialContract` to `fc` or `fincon` in the future. I liked the compactness of emp as a var name

File: packages/serverless-orchestration/test/ServerlessSpoke.js [hunk 10]
```diff
@@ -178,8 +178,8 @@ contract("ServerlessSpoke.js", function(accounts) {
     const invalidConfigResponse = await sendRequest(invalidConfigBody);
     assert.equal(invalidConfigResponse.res.statusCode, 500); // error code
     // Expected error text from an invalid path
-    assert.isTrue(invalidConfigResponse.res.text.includes("Bad environment variables! Specify an `EMP_ADDRESS`"));
-    assert.isTrue(lastSpyLogIncludes(spy, "Bad environment variables! Specify an `EMP_ADDRESS`")); // Check the process logger contained the error.
+    assert.isTrue(invalidConfigResponse.res.text.includes("Bad environment variables! Specify an EMP_ADDRESS"));
+    assert.isTrue(lastSpyLogIncludes(spy, "Bad environment variables! Specify an EMP_ADDRESS")); // Check the process logger contained the error.
```
Comment (daywiss) [inline packages/serverless-orchestration/test/ServerlessSpoke.js:182]: do these errors need to be updated?

**Review by daywiss — COMMENTED** [review]
Hey Chris, great job with this refactor, cant imagine what possessed you to do this though 😄 A lot of it seems like find and replace, but you caught a lot of subtle areas which required manual work.  Review of this is hard because it wont show diffs in areas that were missed, are you fairly certain this covers everything? 

Also does this affect the API of the clients or bots or is it all backwards compat?  I'm thinking about function call names as well as env vars and things like that.

File: packages/financial-templates-lib/src/clients/FinancialContractClient.js (hunk 5 above)
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/FinancialContractClient.js:264]: properly! good catch. will do a find and replace for all `Expiring multi party` and associated changes.

File: packages/financial-templates-lib/src/clients/FinancialContractEventClient.js (hunk 7 above)
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/FinancialContractEventClient.js:22]: done!

File: packages/financial-templates-lib/src/clients/FinancialContractEventClient.js (hunk 8 above)
Comment (chrismaree) [reply packages/financial-templates-lib/src/clients/FinancialContractEventClient.js]: good catch, done! were a few other places with the same string.

File: packages/disputer/test/Disputer.js (hunk 9 above)
Comment (chrismaree) [reply packages/disputer/test/Disputer.js:251]: ye, it's a good point! `fc` seems pretty good actually. Would you like me to do that in this PR or should I do it separately? we would specifically search for the term`financialContract` with and replace with `fc`. would not effect imports or anything like that or any of the other cases like `Financial Contract` used in comments.

WDYT?

File: packages/serverless-orchestration/test/ServerlessSpoke.js (hunk 10 above)
Comment (chrismaree) [reply packages/serverless-orchestration/test/ServerlessSpoke.js:182]: nope.  I think this is correct. I made a small change to the logs that the bots create to accommodate a new (and optional) env variable `FINANCIAL_CONTRACT_ADDRESS` but left the old one as optional to not break the API of the bots configs.

File: packages/disputer/test/Disputer.js (hunk 9 above)
Comment (daywiss) [reply packages/disputer/test/Disputer.js:251]: if its easy for you to do in this one go for it. otherwise its not really that big of a deal

**Review by daywiss — APPROVED** [review]
looks good chris.  going to approve this, up to you if you want to rename the var!

File: packages/serverless-orchestration/test/ServerlessSpoke.js (hunk 10 above)
Comment (mrice32) [reply packages/serverless-orchestration/test/ServerlessSpoke.js:182]: +1

File: yarn.lock [hunk 11]
```diff
@@ -4859,7 +4859,7 @@
     "@uma/common" "^1.1.0"
     "@uma/core-1-1-0" "npm:@uma/core@1.1.0"
 
-"@uma/core-1-2-2@npm:@uma/core@1.2.2", "@uma/core@npm:@uma/core@1.2.2":
+"@uma/core-1-2-2@npm:@uma/core@1.2.2":
```
Comment (mrice32) [inline yarn.lock:4862]: Where did this come from?

File: packages/trader/tsconfig.json (hunk 4 above)
Comment (mrice32) [reply packages/trader/tsconfig.json:4]: +1

File: packages/monitors/README.md [hunk 12]
```diff
@@ -1,6 +1,6 @@
 # @uma/monitors
 
-This package contains four UMA monitor bot implementations: `BalanceMonitor`, `ContractMonitor`, `CRMonitor`, and `SyntheticPegMonitor`. The monitor bots are used to monitor the UMA ecosystem for key events. They have a number of monitor modules built into them that enable real time reporting on key events for a given EMP contract.
+This package contains four UMA monitor bot implementations: `BalanceMonitor`, `ContractMonitor`, `CRMonitor`, and `SyntheticPegMonitor`. The monitor bots are used to monitor the UMA ecosystem for key events. They have a number of monitor modules built into them that enable real time reporting on key events for a given Financial Contract contract.
```
Comment (mrice32) [inline packages/monitors/README.md:3]:
```suggestion
This package contains four UMA monitor bot implementations: `BalanceMonitor`, `ContractMonitor`, `CRMonitor`, and `SyntheticPegMonitor`. The monitor bots are used to monitor the UMA ecosystem for key events. They have a number of monitor modules built into them that enable real time reporting on key events for a given Financial Contract.
```

File: packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js [hunk 13]
```diff
@@ -12,7 +12,7 @@ const {
 const { getTruffleContract } = require("@uma/core");
```
Comment (mrice32) [inline packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js:12]: Are you planning to change the name of this file?

File: packages/financial-templates-lib/test/clients/ExpiringMultiPartyEventClient.js [hunk 14]
```diff
@@ -13,7 +13,7 @@ const {
 const { getTruffleContract } = require("@uma/core");
```
Comment (mrice32) [inline packages/financial-templates-lib/test/clients/ExpiringMultiPartyEventClient.js:13]: Same comment on this file name.

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js (hunk 13 above)
Comment (chrismaree) [reply packages/financial-templates-lib/test/clients/ExpiringMultiPartyClient.js:12]: yup, good catch.

File: yarn.lock (hunk 11 above)
Comment (chrismaree) [reply yarn.lock:4862]: not sure. will revert this one file.

File: packages/affiliates/payouts/dappmining/2021-02-01_2021-02-08_YD-BTC-MAR21_0004.json (hunk 1 above)
Comment (chrismaree) [reply packages/affiliates/payouts/dappmining/2021-02-01_2021-02-08_YD-BTC-MAR21_0004.json:5]: running `yarn lint`

File: packages/disputer/test/Disputer.js (hunk 9 above)
Comment (chrismaree) [reply packages/disputer/test/Disputer.js:251]: for now I'm going to leave it as it is if you ok with that?

**Pull request merged**